lib/CMakeLists.txt

```diff
@@ -96,7 +96,9 @@ add_library(dgllvmpta SHARED
 	llvm/PointerAnalysis/Calls.cpp
 	llvm/PointerAnalysis/Threads.cpp
 )
-target_link_libraries(dgllvmpta PUBLIC dgpta)
+target_link_libraries(dgllvmpta PUBLIC dgpta
+                                # every library should be linked with LLVM
+                                PUBLIC ${llvm})
```

Comment:
I had to make this change to fix dynamic linking with LLVM on macOS. However, I have separated the variables needed for dynamic and static linking with LLVM so that they would not conflict with each other in the future. Therefore, variable `llvm` is set only when linking dynamically and the rest (`llvm_core`, ...) is empty and vice versa. So if there are some problems with static linking, they must be present in `master` as well. 

Thanks for the remark though, I'll add this information to the appropriate `CMakeLists.txt`'s and I'll also test the build with static libraries and if need be, add another commit to this PR.

Reply:
I've added one additional commit that fixed `llvm-sdg-dump` with static LLVM.